elastic/apm-server
# Query cluster UUID on apm-server startup

## Motivation/summary

stack monitoring ui in cloud depends on the beat module of metricbeat, which requires apm-server to know its `cluster_uuid` in order for the documents to be indexed. The apm-server only gets its cluster_uuid as a callback to making an initial connection with elasticsearch.

When using datastreams, the initial connection isn't made until it attempts to index a document, which means that stack monitoring ui is broken for apm-server until a document is indexed, even though nothing may be wrong with the apm-server.

Copy the request made in libbeat and execute it as a precondition to get the cluster_uuid as soon as the server is ready.

## Checklist


## How to test these changes

1. start apm-integration with elasticsearch + kibana
2. install the apm-integration
3. start apm-server: `apm-server -e -E apm-server.data_streams.enabled=true -E http.enabled=true`
4. curl the `/state` internal metrics endpoint: `curl -s http://localhost:5066/state | jq '.outputs.elasticsearch'`
5. you should see non-empty output for the cluster_uuid
